linkerd/app/outbound/src/http.rs

```diff
@@ -22,14 +25,46 @@ mod retry;
 mod server;
 mod strip_proxy_error;
 
-pub use self::logical::Logical;
+pub use self::logical::{profile, LogicalAddr, Routes};
 pub(crate) use self::require_id_header::IdentityRequired;
 pub use linkerd_app_core::proxy::http::{self as http, *};
 
 #[derive(Clone, Debug, PartialEq, Eq, Hash)]
-pub struct Http {
-    target: Logical,
-    version: http::Version,
+pub struct Http<T>(T);
+
+pub fn spawn_routes(
+    mut profile_rx: watch::Receiver<profiles::Profile>,
+    mut mk: impl FnMut(&profiles::Profile) -> Routes + Send + Sync + 'static,
+) -> watch::Receiver<Routes> {
+    let (tx, rx) = {
+        let routes = (mk)(&*profile_rx.borrow());
+        watch::channel(routes)
+    };
+
+    tokio::spawn(async move {
+        loop {
+            tokio::select! {
+                biased;
+                _ = tx.closed() => return,
+                _ = profile_rx.changed() => {
```

Comment:
what, if anything, should this task do if `changed()` returns an error (indicating that the sender of that watch has been dropped)?